docs/performance/inserts/bulk.rst

```diff
@@ -0,0 +1,419 @@
+
+**Shards**
+    Dependant on the number of data-nodes in the cluster (see `CLUSTERED
+    clause`_)
+
+**Replicas**
+    One
+
+You should choose the number of shards wisely. The choice depends on the number
+of nodes in the cluster as well as on the amount of data that is going into the
+table.
+
+Assuming there are six nodes in the cluster, you can put two shards on each
+node, giving you a total of 12 shards. This should be enough to handle millions
+of records.
+
+Shards can be thought of as "virtual nodes." You should create enough for your
+scaling needs, but use as few as possible to keep the resource overhead (e.g.,
+file descriptors and memory) as small as possible.
+
+When importing data, you should set the number of replicas to a minimum,
+ideally zero. If the import fails, you can `drop`_ the table and import
+again. When the import succeeds, adjust the number of replicas according to
+your `availability`_ requirements.
+
+For example, the ``CREATE TABLE`` statement we used before could be changed
+to the following::
+
+    cr> CREATE TABLE users(
+    ...   id INT primary key,
+    ...   name TEXT,
+    ...   day_joined TIMESTAMP,
+    ...   bio TEXT INDEX using fulltext,
+    ...   address OBJECT (dynamic) AS (
+    ...     city TEXT,
+    ...     country TEXT
+    ...   )
+    ... ) CLUSTERED INTO 12 shards
+    ... WITH (number_of_replicas = 0);
+    CREATE OK, 1 row affected (... sec)
+
+
+.. _bulk-refresh-interval:
+
+Refresh interval
+----------------
+
+Another way to speed up importing is to set the `refresh_interval`_ of the
+table to zero::
+
+    cr> ALTER TABLE users SET (refresh_interval = 0);
+    ALTER OK, -1 rows affected (... sec)
+
+This will disable the periodic `refresh`_ of the table that is needed to
+minimize the effect of eventual consistency and also minimize the overhead
+during import.
+
+.. HIDE:
+
+    cr> DROP TABLE users;
+    DROP OK, 1 row affected (... sec)
+
+You can also set the refresh interval when initially creating the table, like
+so::
+
+    cr> CREATE TABLE users (
+    ...   id INT primary key,
+    ...   name TEXT,
+    ...   day_joined TIMESTAMP,
+    ...   bio TEXT INDEX using fulltext,
+    ...   address OBJECT (dynamic) AS (
+    ...     city TEXT,
+    ...     country TEXT
+    ...   )
+    ... ) CLUSTERED INTO 12 shards
+    ... WITH (
+    ...   number_of_replicas = 0,
+    ...   refresh_interval = 0
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+When the import has finished, you can set the refresh interval back to a
+reasonable value (milliseconds)::
+
+    cr> ALTER TABLE users SET (refresh_interval = 1000);
+    ALTER OK, -1 rows affected (... sec)
+
+
+.. _bulk-import-data:
+
+Import the data
+===============
+
+Once the table is created, you can start importing the data.
+
+When importing, CrateDB has `native support for JSON data`_, where each line is
+a JSON string representing a single record. Empty lines are skipped. The keys
```

Comment:
This format is called "JSON Lines" [1], in order to discriminate against regular JSON. Can you add this reference here in a good way?

> JSON Lines files may be saved with the file extension `.jsonl`.

[1] https://jsonlines.org/